src/anm/animation/element.js

```diff
@@ -805,14 +813,13 @@ Element.prototype.render = function(ctx) {
             this.transform(bctx);
             this.painters(bctx);
             this.each(function(child) {
-                child.tick(dt);
                 child.render(bctx, dt);
             });
 
+            // FIXME: this should be performed one time for all masked elements!
             mask.transform(mctx);
```

Comment:
I've made a draft for this, not yet tested: https://github.com/Animatron/player/compare/refactor-timing...refactor-masks. I will ensure if order of rendering (masked elements before mask itself) is right later, but may be you'll see some pitfalls obvious for you.